memphis/memphis.py

```diff
@@ -545,6 +553,45 @@ def parse_descriptor(self, station_name):
         except Exception as e:
             raise MemphisError(str(e)) from e
 
+    async def start_listen_for_functions_updates(self, station_name, first_functions):
+        #first_functions should contain the dict of the first function of each partition key: partition number, value: first function id
+
+        if station_name in self.functions_updates_subs:
+            self.functions_clients_per_station[station_name] += 1
+            return
+        else:
+            self.functions_clients_per_station[station_name] = 1
+
+        functions_updates_subject = "$memphis_functions_updates_" + station_name
+
+        if len(first_functions) == 0:
+            self.functions_updates_data[station_name] = {}
+        else:
+            self.functions_updates_data[station_name] = first_functions
+
+        sub = await self.broker_manager.subscribe(functions_updates_subject)
+        self.functions_updates_subs[station_name] = sub
+
+        loop = asyncio.get_event_loop()
+        task = loop.create_task(
+            self.get_msg_functions_updates(
+                station_name, self.functions_updates_subs[station_name].messages
+            )
+        )
+        self.functions_tasks[station_name] = task
```

Comment:
Please cancel these tasks and stop all functions operation where needed, when destroying producers/stations and on connection close, like we are doing in Schemaverse

Reply:
I handle the destruction in producer.py line: 264